TWLight/emails/backends/mediawiki.py

```diff
@@ -0,0 +1,290 @@
+
+            # Assign the session and email token
+            self.email_token = email_token
+            self.session = session
+            logger.info("Email API session ready.")
+            return True
+        except:
+            if not self.fail_silently:
+                raise
+
+    def close(self):
+        """Unset the session."""
+        self.email_token = None
+        self.session = None
+        logger.info("Session destroyed.")
+
+    def send_messages(self, email_messages):
+        """
+        Send one or more EmailMessage objects and return the number of email
+        messages sent.
+        """
+        if not email_messages:
+            return 0
+        new_session_created = self.open()
+        if not self.session or new_session_created is None:
+            # We failed silently on open().
+            # Trying to send would be pointless.
+            return 0
+        num_sent = 0
+        for message in email_messages:
+            sent = self._send(message)
+            if sent:
+                num_sent += 1
+        if new_session_created:
+            self.close()
+        return num_sent
+
+    @retry_conn()
+    def _send(self, email_message):
+        """A helper method that does the actual sending."""
+        if not email_message.recipients():
+            return False
+
+        try:
+            for recipient in email_message.recipients():
+                # lookup the target editor from the email address
+                target_qs = Editor.objects.values_list("wp_username", flat=True).filter(
+                    user__email=recipient
+                )
+                target_qs_count = target_qs.count()
+                if target_qs_count > 1:
+                    raise Exception(
+                        "Email address associated with {} user accounts, email skipped".format(
+                            target_qs_count
+                        )
+                    )
+
+                target = target_qs.first()
+
+                # GET request to check if user is emailable
+                emailable_params = {
+                    "action": "query",
+                    "list": "users",
+                    "ususers": target,
+                    "usprop": "emailable",
+                    "maxlag": self.maxlag,
+                    "format": "json",
+                }
+
+                logger.info("Checking if user is emailable...")
+                emailable_response = self.session.post(
+                    url=self.url, data=emailable_params
+                )
+                if emailable_response.status_code != 200:
+                    raise Exception(
+                        "There was an error in the request to check if the user can receive emails."
+                    )
+                emailable_data = _json_maxlag(emailable_response)
+                emailable = "emailable" in emailable_data["query"]["users"][0]
+                if not emailable:
```

Comment:
in mediawiki you'll need to add and confirm email addresses for users with wp_usernames that match what's in your TWLight. It's kind of a pain.